src/main/java/io/ballerina/plugins/idea/project/BallerinaProjectUtil.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.ballerina.plugins.idea.project;
+
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+
+import java.io.File;
+import java.nio.file.Paths;
+
+/**
+ * Utility class for ballerina project wise configurations.
+ *
+ * @since 2.0.0
+ */
+public class BallerinaProjectUtil {
+
+    private static final String ballerinaTomlFile = "Ballerina.toml";
+    private static final String ballerinaTestFolderName = "tests";
+    private static final String ballerinaModuleFolderName = "modules";
+
+    public static String findBallerinaPackage(String startingPath) {
```

Comment:
IMO we better use java optional API for this kind of scenarios
```suggestion
    public static Optional<String> findBallerinaPackage(String startingPath) {
```